src/test/scala/wdl4s/expression/FileEvaluatorSpec.scala

```diff
@@ -40,15 +40,26 @@ class FileEvaluatorSpec extends FlatSpec with Matchers {
       WdlSingleFile("3")
     ), WdlMapType(WdlFileType, WdlFileType)),
     (""" [read_string("x"), read_string("y")] """, Seq(WdlSingleFile("x"), WdlSingleFile("y")), WdlArrayType(WdlStringType)),
-    (""" {read_int("a"): read_string("x"), 4: read_string("y")} """, Seq(WdlSingleFile("a"), WdlSingleFile("x"), WdlSingleFile("y")), WdlArrayType(WdlStringType)),
+    (""" [fileNameAsStringInput, "${fileNameAsStringInput}.bai"] """, Seq(WdlSingleFile("sommat.bam"), WdlSingleFile("sommat.bam.bai")), WdlArrayType(WdlFileType)),
+    (""" [ fileNameAsStringInput, mapToFileName["Chris"] ] """, Seq(WdlSingleFile("sommat.bam"), WdlSingleFile("sommatStupid.bam")), WdlArrayType(WdlFileType)),
+    (""" {read_int("a"): read_string("x"), 4: read_string("y")} """, Seq(WdlSingleFile("a"), WdlSingleFile("x"), WdlSingleFile("y")), WdlMapType(WdlIntegerType, WdlStringType)),
     (""" glob("out-*.txt") """, Seq(WdlGlobFile("out-*.txt")), WdlFileType),
     (""" glob("out-*.txt")[0] """, Seq(WdlGlobFile("out-*.txt")), WdlFileType),
-    (""" read_tsv("my_file") """, Seq(WdlSingleFile("my_file")), WdlFileType)
+    (""" read_tsv("my_file") """, Seq(WdlSingleFile("my_file")), WdlFileType),
+    (""" if read_int("i.txt") == 10 then "a.txt" else "b.txt" """, Seq(WdlSingleFile("a.txt"), WdlSingleFile("b.txt"), WdlSingleFile("i.txt")), WdlFileType),
+    (""" if "a" == "b" then "a.txt" else "b.txt" """, Seq(WdlSingleFile("b.txt")), WdlFileType),
```

Comment:
It's the final `WdlFileType` - which in a real output declaration would be passed in based on the declared type of the output